src/common/operator/cast_operators.cpp

```diff
@@ -1409,7 +1409,15 @@ bool TryCastToTimestampNS::Operation(string_t input, timestamp_t &result, bool s
 	if (!TryCast::Operation<string_t, timestamp_t>(input, result, strict)) {
 		return false;
 	}
-	result = Timestamp::GetEpochNanoSeconds(result);
+	if (!Timestamp::IsFinite(result)) {
```

Comment:
Here is an example of testing for infinite values before calling `GetEpochNanoSeconds` in they very PR.